engine/classes/Elgg/Di/DiContainer.php

```diff
@@ -133,6 +131,9 @@ public function setFactory($name, $callable, $shared = true) {
 	 * @throws \InvalidArgumentException
 	 */
 	public function setClassName($name, $class_name, $shared = true) {
+		if (0 === strpos($name, '__')) {
```

Comment:
Yep. substr would be a bit more readable, though most everyday PHP devs recognize common strpos usages I figure.